crates/block-producer/src/transaction.rs

```diff
@@ -0,0 +1,131 @@
+use std::collections::BTreeSet;
+
+use miden_objects::{
+    accounts::AccountId,
+    notes::{NoteId, Nullifier},
+    transaction::{ProvenTransaction, TransactionId, TxAccountUpdate},
+    Digest,
+};
+
+use crate::{errors::VerifyTxError, store::TransactionInputs};
+
+/// A transaction whose proof has been verified.
+#[derive(Clone, PartialEq)]
+pub struct VerifiedTransaction(ProvenTransaction);
```

Comment:
I thought we could use it in the `rpc` component as well to enforce that we verified the proof before sending it onward.

Of course this currently doesn't mean much because the `gRPC` client/server don't enforce this type.

Reply:
I'll remove it for now. If we do want to re-use it in the `rpc` then just it should move to `proto/domain` but that can be part of a greater `proto` type safety effort if/when we do that.